addons/survey/controllers/main.py

```diff
@@ -710,14 +711,67 @@ def _generate_report(self, user_input, download=True):
             ('Content-Disposition', report_content_disposition),
         ])
 
-    def _get_user_input_domain(self, survey, line_filter_domain, **post):
+    def _extract_filters_data(self, survey, post):
+        """
+            Extracts the filters from the URL and returns the related
+            user_input_line_ids and filters data.
+        """
+        # Domain matching all the URL filters
+        filters_domain = []
+        # All suggested answer ids matching the URL filters (for simple and multiple choice questions)
+        suggested_answer_ids = []
+        # Filters data for the front-end display
+        filters_data = []
+
+        for data in post.get('filters', '').split('|'):
+            try:
+                question_type, row_id, answer_id = data.split(',')
+                row_id, answer_id = int(row_id), int(answer_id)
+            except Exception:
+                pass
+            else:
+                if not answer_id:
+                    continue
+                # For every URL filters, build the answer domain
+                answer_domain = []
+
+                if question_type in ('simple_choice', 'multiple_choice'):
+                    answer = request.env['survey.question.answer'].browse([answer_id])
+                    question = answer.question_id
+                    suggested_answer_ids.append(answer_id)
+
+                elif question_type == 'matrix':
+                    answer = request.env['survey.question.answer'].browse([row_id, answer_id])
+                    question = answer[0].matrix_question_id
+                    answer_domain = expression.AND([
+                        ['&', '&', ('matrix_row_id', '=', row_id), ('suggested_answer_id', '=', answer_id), ('question_id', '=', question.id)],
+                        answer_domain
+                    ])
+
+                elif question_type == 'char_box':
```

Comment:
As we are going to change this to support all question types, I will read it more in-depth afterwards.

I can already mention a few things:
- Would be easier to send the question_id than the question_type probably (and then deduct question type from it)

- Using the same key to store suggested_answer and user_input_line models does not seem ideal
Maybe we should try to find a better structure to store things?
I know we are a bit limited since this is all passed into the URL.

One of the way to do it would be to always pass the question_id / user input line, and then deduct the filter from that? Even for single line/multi-line/matrix?
To investigate.